src/main/java/javaslang/control/ValidationBuilder.java

```diff
@@ -0,0 +1,75 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _ _____
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  / /  _  \   Javaslang
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/  \__/  /   Copyright 2014-now Daniel Dietrich
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/___\_/  \_//  \__/_____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+import javaslang.Function2;
+import javaslang.Function3;
+import javaslang.Function4;
+import javaslang.collection.List;
+
+public class ValidationBuilder<E,T1,T2> {
```

Comment:
I would flatten the builders for reasons of simplicity. We may decide in future to increase the number of Tuples. This would lead to an increased depth of Builder classes which is not practical (javadoc for example).